### ISSUE TITLE
NPE in Calculator

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6827]
From Google Play Console for the current beta version (2017.11.01-RC2):

```
java.lang.NullPointerException: 
  at cgeo.geocaching.models.CalcState.<init> (CalcState.java:62)
  at cgeo.geocaching.EditWaypointActivity.fetchCalculatorState (EditWaypointActivity.java:476)
  at cgeo.geocaching.ui.dialog.CoordinatesInputDialog$CalculateListener.onClick (CoordinatesInputDialog.java:578)
  at android.view.View.performClick (View.java:6261)
  at android.widget.TextView.performClick (TextView.java:11180)
  at android.view.View$PerformClick.run (View.java:23748)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Hndler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6776)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1520)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1410)
```

Device info:
Samsung Galaxy S7 and S8 on Android 7.0

Ping @S-Bartfast

**Comment by Lineflyer** [comment]
@S-Bartfast @pstorch This might need your attention before we release a version incl. the calculator

**Comment by Matze2** [comment]
Sorry to interfere, but isn't this duplicate of #6816?

**Comment by pstorch** [comment]
@Matze2 maybe
@Lineflyer which commit is RC2 based on?

**Comment by Lineflyer** [comment]
It is "18:43:43  HEAD is now at ceb2add fix #6816 calculator crash", so #6816 should be fixed (and was tested as fixed by myself before publishing).
Maybe something closely related only occuring under other preconditions?!

**Comment by Matze2** [comment]
The stack above does not match if RC2 contains #6816. I have doubts that the reported version is correct.

**Comment by pstorch** [comment]
This was only a partly fix. I later on found something more and fixed it with commit 5607a2e6d440c940698aca350e3530d2fd2106a5.

**Comment by Lineflyer** [comment]
Sorry, this is the correct stacktrace with slightly other lines in it:

```
java.lang.NullPointerException: 
  at cgeo.geocaching.models.CalcState.<init> (CalcState.java:62)
  at cgeo.geocaching.EditWaypointActivity.fetchCalculatorState (EditWaypointActivity.java:476)
  at cgeo.geocaching.ui.dialog.CoordinatesInputDialog$CalculateListener.onClick (CoordinatesInputDialog.java:578)
  at android.view.View.performClick (View.java:6213)
  at android.widget.TextView.performClick (TextView.java:11074)
  at android.view.View$PerformClick.run (View.java:23645)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6692)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1468)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1358)
```

**Comment by Lineflyer** [comment]
> This was only a partly fix. I later on found something more and fixed it with commit 5607a2e.

Do you think the current content of `release` is ready to push another beta version...or first wait on leading/trailing zero finalization?

**Comment by Matze2** [comment]
EditWaypointActivity.java:476 does not contain a constructor anymore with #6816.

**Comment by pstorch** [comment]
> Do you think the current content of release is ready to push another beta version...or first wait on leading/trailing zero finalization?
I think it's ok for another RC. But for the final release the leading/trailing zero change should be in.

**Comment by Lineflyer** [comment]
No more appearance of this crash with 2017.11.07-RC
